### PR TITLE
Introduce `default-members` to the workspace root

### PR DESCRIPTION
**Pull request by s0me0ne-unkn0wn**
Issue: #1418

This PR aims only to build the Polkadot node software by default when `cargo build` is run from the repo root, avoiding building tons of packages not necessary to build and run the node.

Must be followed up with proper documentation changes (see #1372 for example; other use cases must be described, too).

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -452,6 +452,7 @@ members = [
 	"substrate/utils/prometheus",
 	"substrate/utils/wasm-builder",
 ]
+default-members = [ "polkadot" ]
```
Comment (ggwpez) [inline Cargo.toml:455]: Will this also build the needed execution worker binaries?

Comment (s0me0ne-unkn0wn) [reply Cargo.toml:455]: Yes, they are all binaries of the `polkadot` package. The only thing broken here is `cargo run` (which would only build `polkadot` binary but not workers), but it's been broken for some time anyway.

**Review by ggwpez — APPROVED** [review]

**Comment by mrcnski** [comment]
This is totally random and probably the wrong place to ask it, but are the workers inheriting the profile values correctly. I just build them with `production` profile but [list-syscalls.rb](https://gist.github.com/koute/166f82bfee5e27324077891008fca6eb) doesn't work (it did before binary separation). I'm guessing the profile is not applied, particularly the `lto = true`.

File: Cargo.toml (hunk 1 above)
Comment (bkchr) [reply Cargo.toml:455]: We should also set default run

Comment (s0me0ne-unkn0wn) [reply Cargo.toml:455]: Makes sense. I believe it selects the first binary anyway, but it's always good to be explicit.

**Comment by s0me0ne-unkn0wn** [comment]
@mrcnski I've caught some output from `cargo build --release --verbose` and formatted it so it's easy to compare just scrolling the lines, it's `polkadot` binary and one of the workers. I see no mentions of `-C lto` here, so chances are it's not propagated properly.

```
     Running `/home/s0me0ne/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/rustc --crate-name polkadot                --edition=2021 polkadot/src/main.rs               --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=316 --crate-type bin --emit=dep-info,link -C opt-level=3 -C embed-bitcode=no -C metadata=57b404de214ca524 -C extra-filename=-57b404de214ca524 --out-dir /home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps -L dependency=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps --extern color_eyre=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps/libcolor_eyre-f93ed37c7757fb41.rlib --extern polkadot_cli=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps/libpolkadot_cli.rlib --extern polkadot_node_core_pvf=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps/libpolkadot_node_core_pvf-4527c8ad49d0dbb8.rlib --extern polkadot_node_core_pvf_common=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps/libpolkadot_node_core_pvf_common-c81b0c7912854902.rlib --extern polkadot_node_core_pvf_execute_worker=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps/libpolkadot_node_core_pvf_execute_worker-1f84d1b1c9059842.rlib --extern polkadot_node_core_pvf_prepare_worker=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps/libpolkadot_node_core_pvf_prepare_worker-ca5fa595684db984.rlib --extern polkadot_overseer=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps/libpolkadot_overseer-6a7b44e9bf2d1f86.rlib --extern tikv_jemallocator=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps/libtikv_jemallocator-d9ac9cb80955f730.rlib -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/secp256k1-sys-246fe31b64b3f8a4/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/psm-e3ae4e50088c06e9/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/zstd-sys-7f24cdb9fc33d6f9/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/wasmtime-runtime-4769ec7f9d73c093/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/blake3-c34f860f630619ad/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/blake3-c34f860f630619ad/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/librocksdb-sys-36eae9a495fe1557/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/librocksdb-sys-36eae9a495fe1557/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/tikv-jemalloc-sys-c02e2a4d456b1802/out/build/lib -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/lz4-sys-a151bbd581a19649/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/ring-18e6a20cc221ff7e/out`
     Running `/home/s0me0ne/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/rustc --crate-name polkadot_execute_worker --edition=2021 polkadot/src/bin/execute-worker.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=316 --crate-type bin --emit=dep-info,link -C opt-level=3 -C embed-bitcode=no -C metadata=c78b1ea328e9720f -C extra-filename=-c78b1ea328e9720f --out-dir /home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps -L dependency=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps --extern color_eyre=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps/libcolor_eyre-f93ed37c7757fb41.rlib --extern polkadot_cli=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps/libpolkadot_cli.rlib --extern polkadot_node_core_pvf=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps/libpolkadot_node_core_pvf-4527c8ad49d0dbb8.rlib --extern polkadot_node_core_pvf_common=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps/libpolkadot_node_core_pvf_common-c81b0c7912854902.rlib --extern polkadot_node_core_pvf_execute_worker=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps/libpolkadot_node_core_pvf_execute_worker-1f84d1b1c9059842.rlib --extern polkadot_node_core_pvf_prepare_worker=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps/libpolkadot_node_core_pvf_prepare_worker-ca5fa595684db984.rlib --extern polkadot_overseer=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps/libpolkadot_overseer-6a7b44e9bf2d1f86.rlib --extern tikv_jemallocator=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/deps/libtikv_jemallocator-d9ac9cb80955f730.rlib -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/secp256k1-sys-246fe31b64b3f8a4/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/psm-e3ae4e50088c06e9/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/zstd-sys-7f24cdb9fc33d6f9/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/wasmtime-runtime-4769ec7f9d73c093/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/blake3-c34f860f630619ad/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/blake3-c34f860f630619ad/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/librocksdb-sys-36eae9a495fe1557/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/librocksdb-sys-36eae9a495fe1557/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/tikv-jemalloc-sys-c02e2a4d456b1802/out/build/lib -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/lz4-sys-a151bbd581a19649/out -L native=/home/s0me0ne/wrk/parity/polkadot-sdk/target/release/build/ring-18e6a20cc221ff7e/out`
```

**Review by chevdor — APPROVED** [review]

**Review by mrcnski — APPROVED** [review]
Nice!

About my issue, looks like `-C lto` is getting propogated for `--profile production`.

**Review by coderobe — APPROVED** [review]

**Pull request merged**